src/utils/helper.js

```diff
@@ -457,15 +455,15 @@ export const updateInternalQuery = (
 	}
 };
 // extracts query options from defaultQuery if set
-export const extractQueryFromDefaultQuery = (defaultQuery) => {
+export const extractQueryFromDefaultQuery = (defaultQuery, props) => {
 	let queryToBeReturned = {};
 	if (defaultQuery) {
 		const evaluateQuery = defaultQuery();
 		if (evaluateQuery) {
 			// we should only retrieve and set the query options here.
 			// [Not implemented yet] `query` key should be handled separately for
 			// adding it to `queryList` in the redux store
-			const { query, ...options } = evaluateQuery;
+			const { query, ...options } = evaluateQuery([], props);
```

Comment:
@ShahAnuj2610 Why the value is an empty array?
Use the `current value` which is specific to the component.